types.go

```diff
@@ -153,14 +153,17 @@ var doubleEffect = map[Type]Type{
 }
 
 // Get effectiveness of a given elemental type matchup.
-func GetElementalEffect(move, def Type) Effectiveness {
-	if noEffect[move]&def > 0 {
-		return NoEffect
-	}
-
+func GetElementalEffect(move, def Type, item Item) Effectiveness {
 	reduce := bits.OnesCount32(uint32(halfEffect[move] & def))
 	increase := bits.OnesCount32(uint32(doubleEffect[move] & def))
 	effect := increase - reduce
+	if noEffect[move]&def > 0 {
+		if move&TypeGround > 0 && def&TypeFlying > 0 && item == ItemIronBall {
+			// Flying immunity negated
+		} else {
+			return NoEffect
+		}
+	}
```

Comment:
This should be handled in the `EffectiveType` getter method for the pokemon

Reply:
I don't think this works because we can't change the type of the Pokemon in this case, we only want to remove it's immunity. If we change the Flying type to something like normal, then we introduce a super-effective Fighting matchup that is not expected.

Reply:
I see. I would still prefer this implementation not modify `GetElementalEffect`. Looking at bulbapedia, holding an iron ball applies [Grounded](https://bulbapedia.bulbagarden.net/wiki/Grounded) to the pokemon. We should have a helper function to determine if a pokemon is grounded. Then, we should have a special condition in `CalcMoveDamage` to handle that case, another helper function to handle all the special cases around type effectiveness.